lib/net.js

```diff
@@ -2019,6 +2020,10 @@ Server.prototype.listen = function(...args) {
     toNumber(args.length > 2 && args[2]);  // (port, host, backlog)
 
   options = options._handle || options.handle || options;
+  if (typeof options.host === 'string' && RegExpPrototypeExec(/^[a-zA-Z0-9-:.]+$/, options.host) === null) {
```

Comment:
```suggestion
  if (typeof options.host === 'string' && RegExpPrototypeExec(/^[a-zA-Z0-9-:%.]+$/, options.host) === null) {
```

For scoped literal IPv6 addresses.

Reply:
Also maybe store the regex as a variable / add a comment about it